### PR TITLE
Simplify system controller and flesh out tests

### PR DESCRIPTION
**Pull request by ddollar**

### CODE REVIEW

**Comment by ddollar** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;LmWtL46b&#x2F;597-provider-system"> provider: system</a>

File: api/controllers/system.go [hunk 1]
```diff
@@ -43,61 +23,37 @@ func SystemUpdate(rw http.ResponseWriter, r *http.Request) *httperr.Error {
...
-			if count <= 1 {
-				return httperr.Errorf(403, "count must be greater than 1")
-			}
-
-			stack, err := models.Provider().AppGet(rack.Name)
-			if err != nil {
-				return httperr.Errorf(404, "no such stack")
-			}
-
-			if stack.Parameters["Autoscale"] == "Yes" {
```
Comment (MiguelMoll) [inline api/controllers/system.go]: Is this being checked elsewhere? I remember there being issues trying to set count with autoscale enabled.

Comment (ddollar) [inline api/controllers/system.go]: Added another check to replace this, thanks :)

**Comment by ddollar** [comment]
Need to re-add the rack:update notification when things change in provider.SystemSave

**Pull request closed without merging**